### ISSUE TITLE
travis ci failing with advent of ipython 6.0.0rc1

### ISSUE THREAD

**Issue by nealmcb** [google/python-fire#57]
ipython 6.0.0rc1 was released 7 hours ago:

* https://github.com/ipython/ipython/releases

Now when building fire, travis CI finds that version, which doesn't support Python 2.7, and complains, e.g. at https://travis-ci.org/nealmcb/python-fire/jobs/220816890

```
...
Processing dependencies for fire==0.1.0
Searching for ipython
Reading https://pypi.python.org/simple/ipython/
Downloading https://pypi.python.org/packages/76/46/c505c48d264715136edde1a1960f9158133e34c03bbfe4c955dbc5948ce4/ipython-6.0.0rc1.tar.gz#md5=e57acb5fabd9f18b739de35a1f00eac5
Best match: ipython 6.0.0rc1
Processing ipython-6.0.0rc1.tar.gz
Writing /tmp/easy_install-Aw9UOD/ipython-6.0.0rc1/setup.cfg
Running ipython-6.0.0rc1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-Aw9UOD/ipython-6.0.0rc1/egg-dist-tmp-JVscrS
IPython 6.0+ does not support Python 2.6, 2.7, 3.0, 3.1, or 3.2.
...
error: Setup script exited with 1
The command "python setup.py develop" failed and exited with 1 during .
```

Time for a dependency change. For a lot of projects....

**Comment by nealmcb** [comment]
Looks like `'ipython<6.0'` in setup.py will do the trick.

**Comment by jtratner** [comment]
what's strange is that pip is supposed to ignore it (according to IPython's readme). I'm wondering if the IPython project messed up the trove classifiers or pip did something wrong

**Comment by Carreau** [comment]
Sorry about that. I'm confused how you get rc1 without using the --pre flag. Or am I missing a configuration option. We'll improve the readme and the docs for downstream projects. Thanks a lot for catching this an warning us upstream.




> I'm wondering if the IPython project messed up the trove classifiers or pip did something wrong

The trove classifiers are purely informal. We (the ipython team) had to send patch upstream to Pip, Warehourse, and PyPI, plus modify some peps, in order for the `requires_python > 3.3` metadata to be understood by setuptools/pip.. etc. We likely missed something in setuptools.

**Comment by jtratner** [comment]
Thanks @carreau - there are a bunch of moving parts here to take care of
(and I'm sure it simplifies things for you folks to be python 3.3+ only :) )

An IPython dev suggested a workable way to do conditional requirement for
IPython here that should work for us -
https://github.com/ipython/ipython/issues/10329#issuecomment-293294424

**Comment by Carreau** [comment]
Thanks. I'll update the IPython readme with the information Thomas proposal. 

> (and I'm sure it simplifies things for you folks to be python 3.3+ only :) )

Yes, yes.. I hope to be 3.4+ soon to have the `async`/`await` syntax. This is one of the first IPython releases that will have less lines (-1.5k lines) than the previous one. And types annotations are really great even just as documentation.

**Comment by ebutler-allies** [comment]
6.0.0 just landed less than 30 minutes ago, and killed one of my projects build jobs. I see a merged branch, but there doesn't appear to be a cut release on pypi with that fix. What's the ETA?

**Comment by dbieber** [comment]
Closing since we currently list 'ipython<6.0' as our dependency and travis is happy.
Once we close out #7 this will be even cleaner since we won't list ipython as a required dep at all.

**Comment by wookayin** [comment]
PR was merged in #58

**Comment by dbieber** [comment]
#7 is now done as well, so ipython is no longer listed as a dependency.
